### PR TITLE
Upgrade to otj-parent-spring

### PR DESCRIPTION
**Pull request by SegFaultAX**
* Latest otj-parent-spring (179)
* Fix PMD violations
  * Make fully static classes utility (final + private ctor)
  * Minimize usage of `instanceof` inside of exceptions
  * Suppress things that ought to be suppressed

### CODE REVIEW

File: src/main/java/com/opentable/util/ExceptionSwallower.java [hunk 1]
```diff
@@ -43,8 +47,6 @@ public static Runnable swallowExceptions(Runnable in)
                 LOGGER.error("Uncaught exception swallowed", t);
                 if (t instanceof Error) {
                     throw t;
-                } else if (t instanceof InterruptedException) {
```
Comment (SegFaultAX) [inline src/main/java/com/opentable/util/ExceptionSwallower.java]: This condition is impossible.

**Comment by mikebell90** [comment]
This cannot be merged I will explain later

File: src/main/java/com/opentable/callback/ExecutorBatchingCallback.java [hunk 2]
```diff
@@ -93,8 +93,8 @@ public void call(final List<T> item) throws Exception
             inFlight.incrementAndGet();
             executor.submit(new ExecutorCallable<T>(out, item));
 
-            Future<Void> f;
-            while ( (f = executor.poll()) != null ) {
+            Future<Void> f = executor.poll();
+            while (f != null) {
```
Comment (SegFaultAX) [inline src/main/java/com/opentable/callback/ExecutorBatchingCallback.java]: Open to other suggestions on how to unwrap the assignment-in-loop-condition. (Can also just suppress it out)

File: src/main/java/com/opentable/util/ExceptionSwallower.java (hunk 1 above)
Comment (mikebell90) [reply src/main/java/com/opentable/util/ExceptionSwallower.java]: How is that impossible. You catch throwable. And remember people can sneakythrows and stuff like that

**Pull request closed without merging**